### ISSUE TITLE
Misleading names for context parameters exposing the number of logical views and actual views

### ISSUE THREAD

**Issue by javaserverfaces** [javaee/mojarra#1900]
The JSF Sun RI provides 2 context parameters used to set the number of logical views and actual views stored when using server-side state saving:

*   com.sun.faces.numberOfLogicalViews
*   com.sun.faces.numberOfViewsInSession

These parameters are very important as they have an impact on the amount of memory allocated per user's session.

The description of these parameters is not easy to find, one description can be found there :
[http://wikis.sun.com/display/GlassFish/JavaServerFacesRI](http://wikis.sun.com/display/GlassFish/JavaServerFacesRI)

The description states :
"com.sun.faces.numberOfViewsInSession - Defaults to 15\. This parameter defines the maximum number of JSF views stored in the session for per logical view. The map used to store the views uses an LRU algorithm to keep the map from growing beyond the configured value."
"com.sun.faces.numberOfLogicalViews - Defaults to 15\. This parameter defines the maximum number of logical views to store per session. The map used to store the logical views uses an LRU algorighm to keep the map from growing beyone the configured value."

JBoss documentation goes in the same direction : [http://docs.jboss.org/jbossas/6/JSF_Guide/en-US/html/jsf.reference.html](http://docs.jboss.org/jbossas/6/JSF_Guide/en-US/html/jsf.reference.html)

"com.sun.faces.numberOfViewsInSession : For server state-saving, how many views, per logical view, can be stored in the session before oldest is removed?"
"com.sun.faces.numberOfLogicalViews : For server state-saving, how many logical views are allowed before oldest is removed?"

These descriptions made me think the logical views contain views and the number of logical views is defined by com.sun.faces.numberOfLogicalViews.
However, after experiencing problems during some tests, we found out that the size of the logical map is equal to the parameter com.sun.faces.numberOfViewsInSession and the size of the view map is equal to com.sun.faces.numberOfLogicalViews.
The code in charge of creating the different maps can be found in com.sun.faces.application.StateManagerImpl.

In a nutshell, I have the feeling that the two context parameters have been inverted in the State manager.
Is it a a bug, a documentation problem or just me not understanding the documentation ?
#### Affected Versions
[2.0, 2.1]
